monkey/monkey_island/cc/services/authentication_service/refresh_token_provider.py

```diff
@@ -0,0 +1,31 @@
+from typing import TypeAlias
+
+from flask_security import Security
+
+RefreshToken: TypeAlias = str
+
+
+class RefreshTokenManager:
+    def __init__(self, security: Security):
+        self._token_serializer = security.remember_token_serializer
+        self._refresh_token_expiration = (
+            security.app.config["SECURITY_TOKEN_MAX_AGE"]
+            + security.app.config["SECURITY_REFRESH_TOKEN_TIMEDELTA"]
+        )
+
+    def generate_refresh_token(self, payload: str) -> RefreshToken:
+        """
+        Generates a refresh token for a user
+        :param payload: String that will be encoded in the token
+        :return: A refresh token
+        """
+        refresh_token = self._token_serializer.dumps(payload)
+
+        return refresh_token
+
+    def validate_refresh_token(self, refresh_token: str):
```

Comment:
When do we need to validate the refresh token ? Shouldn't we also do it after generating? Or is this check for the expiration of the refresh token ?

Reply:
1. We need to validate the refresh token when a user attempts to refresh a token.
2. What's the point of validating it after generation? We know we generated a valid token
3. Validation includes the check for expiration. But maybe I should add a test for a malformed token